kubeflow/trainer/rhai/traininghub_instrumentation.py

```
@@ -0,0 +1,961 @@
+                        )
+                    else:
+                        print("[Kubeflow] No stale metrics files found", flush=True)
+
+                except OSError as e:
+                    print(f"[Kubeflow] Warning: Metrics cleanup failed: {e}", flush=True)
+        else:
+            print("[Kubeflow] Non-primary pod skipping metrics cleanup", flush=True)
+
+        # Start HTTP server for metrics
+        try:
+            server = http.server.HTTPServer(("0.0.0.0", metrics_port), TrainingHubMetricsHandler)
```

Ruff (0.16.0)

[error] Possible binding to all interfaces

(S104)